libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c

```diff
@@ -412,11 +412,6 @@ int32_t SOCKETS_Recv( Socket_t xSocket,
 {
     ss_ctx_t * ctx = ( ss_ctx_t * ) xSocket;
 
-    if( ( ctx->status & SS_STATUS_CONNECTED ) != SS_STATUS_CONNECTED )
-    {
-        return SOCKETS_ENOTCONN;
-    }
-
     if( SOCKETS_INVALID_SOCKET == xSocket )
```

Comment:
Note that `SOCKETS_INVALID_SOCKET != NULL`, (definition [here](https://github.com/aws/amazon-freertos/blob/master/libraries/abstractions/secure_sockets/include/iot_secure_sockets.h#L110)) so the commit message is incorrect.  Let's change the commit message to "Validate socket handle before dereferencing."